### PR TITLE
fix: support new is_my_account() method from main plugin

### PR DESCRIPTION
**Pull request by dkoo**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-theme/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Currently, transactions completed from the "My Account" dashboard are not using the modal checkout flow. However, due to a fix in #1573 for express checkout flows, when completing a transaction from My Account that's also related to a subscription that originated with a modal checkout flow, these transactions are flagged as My Account transactions and show the modal checkout thank you template, but in a non-modal context, trapping the user in a dead-end.

This fix relies on a new method in https://github.com/Automattic/newspack-plugin/pull/3389 that lets us determine if a checkout flow originated from My Account, and show the correct non-modal thank you template if so.

The better solution would be to use the modal checkout flow for My Account transactions too, but this is a bigger change than is appropriate for a hotfix.

### How to test the changes in this Pull Request:

See testing instructions in https://github.com/Automattic/newspack-plugin/pull/3389.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-modal-checkout.php [hunk 1]
```diff
@@ -124,34 +124,37 @@ public static function woocommerce_checkout_create_order_line_item( $item, $cart
...
 	public static function woocommerce_payment_successful_result( $result ) {
 		$order_id           = $result['order_id'];
 		$order              = \wc_get_order( $order_id );
 		$modal_checkout_url = $order->get_meta( '_newspack_modal_checkout_url' );
-		if ( empty( $modal_checkout_url ) ) {
+		if (
+			empty( $modal_checkout_url ) ||
+			(
+				// Until we use the modal checkout flow from My Account, we don't want to show the modal checkout thank you template for checkouts originating from My Account.
+				method_exists( 'Newspack\WooCommerce_My_Account', 'is_from_my_account' ) && \Newspack\WooCommerce_My_Account::is_from_my_account()
```
Comment (adekbadek) [inline includes/class-modal-checkout.php]: When following the testing instructions, the successful result is never referred from my-account, but from /checkout. That might be why I'm not seeing what's expected here: https://github.com/Automattic/newspack-plugin/pull/3389#pullrequestreview-2279254728

Comment (dkoo) [reply includes/class-modal-checkout.php]: The hidden field [added here](https://github.com/Automattic/newspack-plugin/pull/3389/files#diff-74c4e95b1dc2575ce46450e3fa231d01a291e298a72ed7656a6271757eb585afR500-R505) should account for that. However, it won't work for express checkouts, and I'm not sure how we might pass that info along to express checkout requests...

Comment (dkoo) [reply includes/class-modal-checkout.php]: Fixed by [146917f](https://github.com/Automattic/newspack-plugin/commit/146917fb25b1b36368fe92f50adc9c2574330f6f)

**Comment by dkoo** [comment]
Marking approved based on @adekbadek's [approval in the other PR](https://github.com/Automattic/newspack-plugin/pull/3389#pullrequestreview-2302561432).

**Comment by ronchambers** [comment]
New bug posted here: https://github.com/Automattic/newspack-plugin/issues/3426

**Pull request merged**